mobify/platform-scaffold
# [WEB-1393] Performance Fix: Delay `getCart`

In an effort to avoid unnecessary network requests, I have delayed fetching for cart content (i.e. content for the mini cart) until after the user scrolls. There are other actions that trigger fetching of cart content, such as opening the `miniCart`, adding products to the cart, etc.

 **JIRA**: [WEB-1393](https://mobify.atlassian.net/browse/WEB-1393)

## Changes
- Prevent `getCart` from being called on initial page load
- Add window scroll handler to run `getCart`
- Add placeholders into the MiniCart when cart content hasn't been loaded yet

## How to test-drive this PR
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Visit any product page and add it to the cart
- Return to the homepage and refresh the page
- After the site reloads, notice that the mini-cart does not get updated to reflect that you have an item in the cart
- Scroll down the page
- **Verify** that after scrolling does the mini cart update

**Bonus**: verify the above steps for Salesforce too!

## Results

| Load Cart only on Scroll | Placeholders in `MiniCart` |
| --- | --- |
| ![cart-load](https://user-images.githubusercontent.com/734535/27112660-48c730c8-506d-11e7-8503-914c8a1c3370.gif) | ![cart-load-placeholder](https://user-images.githubusercontent.com/734535/27112661-4c520b8c-506d-11e7-9d46-70057a3ad454.gif) |
